Feature/style cleanup

SeanScripts opened the pull request:
Lint fixes for: 
- [C0411](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/wrong-import-order.html) (wrong-import-order)
- [R1705](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html) (no-else-return)
- [R1720](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-raise.html) (no-else-raise)

Greenford commented:
Apologies, @SeanScripts. [R1705](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html) (no-else-return) and [R1720](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-raise.html) (no-else-raise) were generated before you knew to use the [.pylintrc](https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/main/.pylintrc). My bad. 

From @kmaphoenix : "Yeah I think he was using a different linter. Because I've linted the whole library without those errors"

Yes, while the edits (I haven't but glanced at them) supposedly won't alter our execution flow if we completely trusted pylint, we've opted to adopt a "if it ain't broken, don't fix it" mentality, as this allows us to avoid testing each change. Much less work for all of us. 

Takeaways: Do use our specific .pylintrc. I'm partially repeating that for myself. 

Hopefully one of @kmaphoenixm @MRyderOC @cgibson6279 or myself can give a LGTM on the import fix after looking at the [C0411](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/wrong-import-order.html) (wrong-import-order) commit. 

In the meantime, feel free to make more style edits (using our specific .pylintrc) on this branch.

SeanScripts commented:
Yes, I was using the default pylint settings, sorry. I didn't know about pylintrc. Though it looks like the .pylintrc file has the wrong-import-order convention disabled too, anyway.

SeanScripts commented:
I modified the .pylintrc file to remove the deprecated options files-output and no-space-check (which we weren't even using). Hopefully this is fine. It just removes the deprecation warnings that show up every time pylint is run with this settings file.

SeanScripts commented:
These are the remaining lint fixes to get everything up to 10/10, other than core/intents.py, which has one section that pylint thinks has too many nested blocks, but which isn't really avoidable.

SeanScripts commented:
These changes are only making the docstrings follow a more consistent format, with one exception:
In the core_async/test_cases.py file, I updated the end of the batch_delete_test_cases function.
batch_delete_test_cases in the CX API returns None, so I updated it to not store/return the results (which would be None)

The format I chose for the docstrings was just the one that I saw most often in the project. I also updated the docstrings to not include the name of the return value, as was mentioned in sub-issue #2 in the open issue #29 for Repo-wide style cleanup.

kmaphoenix commented on src/dfcx_scrapi/core/experiments.py:
```diff
@@ -56,7 +56,15 @@ def __init__(
     def list_experiments(self, environment_id=None):
-        """list out experiments"""
+        """List out experiments.
+        
```
whitespace

kmaphoenix commented on src/dfcx_scrapi/core/flows.py:
```diff
@@ -309,16 +311,16 @@ def import_flow(
-          lro.result, If successful the LRO result will return the Flow ID of
-            the newly imported Flow. Otherwise, it will return the
-            corresponding error.
+          A Long Running Operation result. If successful the LRO result will 
```
check whitespace

kmaphoenix reviewed (changes requested):
Since this is an XL change (many files) let's limit the scope of what we're changing to the following items:
* Docstrings
* Import ordering
* Whitespace / Spacing

Please revert any change to a method for this branch, and address in a separate branch with testing, etc.

Additionally, you'll want to run the linter over all the files again.  
Adding newlines in several places introduced more whitespace, which was throwing warnings in pylint

SeanScripts commented on src/dfcx_scrapi/builders/intents.py:
```diff
@@ -211,7 +211,7 @@ def show_intent(
         elif mode in ["phrases", "training phrases"]:
             print(self._show_training_phrases(repeat_count=repeat_count))
         elif mode == "whole":
-            print(self.__str__())
```
self.__str__() and str(self) are the same, and __str__ is just the override for the string conversion. This was what the pylint warning for this file was.

SeanScripts commented on src/dfcx_scrapi/core/conversation.py:
```diff
@@ -313,8 +321,8 @@ def _get_intent_detection(self, test_set: pd.DataFrame):
             )
             threads[i].start()
 
-        for idx, _ in enumerate(threads):
```
This was a lint fix. The original code was:
```
for idx, _ in enumerate(threads):
  threads[idx].join()
```
This iterates over the list of threads and runs the `join` method for each one. Using `enumerate` and then ignoring the object, only to use the index to get the object back, is the unnecessary part here that the linter pointed out.
This is equivalent to:
```
for thread in threads:
  thread.join()
```
Which I think you'll agree is simpler conceptually and visually, and should be functionally identical. However, I'll still look into testing it.

SeanScripts commented on src/dfcx_scrapi/core/entity_types.py:
```diff
@@ -187,14 +212,14 @@ def entity_types_to_df(
         Returns:
-          - intents_map, Dictionary containing Entity UUIDs as keys and
+          Dictionary containing Entity UUIDs as keys and
               intent.display_name as values
```
Good catch, I missed that one

SeanScripts replied:
Also noticed this one said Entity rather than Entity Type, which may have been a bit confusing, since it contains the entity type names and UUIDs, and not the entities within them.

kmaphoenix replied:
@SeanScripts Ack. `Entity Type` is the correct semantics to use here.

SeanScripts commented on src/dfcx_scrapi/core/agents.py:
```diff
@@ -83,6 +83,7 @@ def list_agents(
               `projects/<GCP PROJECT ID>/locations/<LOCATION ID>`
               `projects/my-gcp-project/locations/us-central1`
           project_id: The GCP Project ID as a string
+        
```
Yep. Fixed those.

kmaphoenix commented on src/dfcx_scrapi/tools/validation_util.py:
```diff
@@ -75,7 +76,7 @@ def validation_results_to_dataframe(self, validation_results: Dict):
                 temp_df.insert(0, "flow", flows_map[temp])
 
-                max_cols_new = max([len(x) for x in temp_df.resourceNames])
+                max_cols_new = max(len(x) for x in temp_df.resourceNames)
```
@SeanScripts as mentioned above, I'll sync w/you re: `.pylintrc` file and contents.  

Admittedly, we're behind on implementing automated regression testing for the repo, which we should definitely do ASAP to streamline changes like this.

Until then, if you don't mind, please move these changes to a smaller PR where we can provide more documentation on regression testing.

kmaphoenix commented:
@SeanScripts Additionally, please perform a `rebase` on your local copy with `main`.  
There were some changes in `sessions.py` that are in conflict now.

kmaphoenix reviewed (approved):
LGTM
